File: ch2/src/cpp_demo_pkg/src/person_node.cpp
#include <iostream>
#include <string>
#include <rclcpp/rclcpp.hpp>

class PersonNode: public rclcpp::Node {
private:
    std::string name_;
    int age_;

public:
    PersonNode(const std::string& node_name, const std::string name, const int age)
    : Node(node_name), name_(name), age_(age) {};

    void eat(const std::string& food) {
        RCLCPP_INFO(this->get_logger(), "%s, %d years old, eating %s now.", this->name_.c_str(), this->age_, food.c_str());
    }
};

int main(int argc, char** argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("cpp_node", "chenyinjie", 18);
    node->eat("apple");
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}

File: ch2/src/py_demo_pkg/py_demo_pkg/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self, node_name:str, name:str, age:int) -> None:
        super().__init__(node_name)
        self.name = name
        self.age = age
        
    def eat(self, food:str):
        self.get_logger().info(f"my name is {self.name}, {self.age} years old, and eating {food} now.")
        

def main():
    rclpy.init()
    node = PersonNode('person_node', 'chenyinjie', 18)
    node.eat('apple')
    rclpy.spin(node)
    rclpy.shutdown()